ob1k-core/src/main/java/com/outbrain/ob1k/client/endpoints/AsyncClientEndpoint.java

```diff
@@ -83,15 +88,52 @@ public AsyncClientEndpoint(final HttpClient httpClient, final RequestMarshallerR
     }
   }
 
+  public class InvokeAsyncAction<T> implements FutureAction<T> {
```

Comment:
it is public because it's in use in other places? if yes, please extract to a file, else make it private